app/product-management/js/views/part/part_list_item.js

```diff
@@ -178,6 +182,347 @@ define([
+                                rootContext.buildChildrenNodes(data, includeUl);
+                                if ($(item).is(':last-child')) {
+
+                                    item.setAttribute('class','lastExpandable embeddedExpandable');
+                                    includeDiv.setAttribute('class', 'blockClickAreaEmbedded hitarea lastExpandable-hitarea expandable-hitarea embeddedExpandable');
+                                }
+                                includeUl.setAttribute('style', 'display:none;');
+                                item.append(includeUl);
+                                item.prepend(includeDiv);
+                            }
+                        }
+                    });
+                });
+            }
+        },
+        buildChildrenNodes: function (data, listChild) {
+
+            var lastIteration = data.lastIterationNumber - 1;
+            var assemblyArray = data.partIterations[lastIteration].components;
+            var rootContext = this;
+            assemblyArray.forEach(function (element, index) {
+
+                rootContext.getChildLatestVersion(data.workspaceId, element.component.number).success(function (latest) {
+
+                    var fullId = 'path_-1-' + element.fullId;
+                    var item = document.createElement('li');
+                    var span = document.createElement('span');
+                    span.setAttribute('class', 'product_title');
+                    $(span).text(latest.partKey);
+                    item.append(span);
+                    rootContext.buildHierarchy(element.component.number, item, fullId, data.workspaceId);
+                    if (index === (assemblyArray.length - 1)) {
+
+                        var lastestIteration = latest.lastIterationNumber - 1;
+                        var components  = latest.partIterations[lastestIteration].components;
+                        if(components.length !== 0){
+
+                            item.setAttribute('class', 'lastExpandable embeddedExpandable');
+                            var divIntoItem = $(item).children('.blockClickAreaEmbedded');
+                            divIntoItem.addClass('lastExpandable-hitarea');
+                        }else{
+
+                            item.setAttribute('class', 'last');
+                        }
+                    }
+                    listChild.append(item);
+                });
+            });
+        },
+        buildHierarchy: function (partNumber, currentLine, parentId, workspaceId) {
+
+            var rootContext = this;
+            this.getChildLatestVersion(workspaceId, partNumber).success(function (latestChild) {
+
+                var lastIteration = latestChild.lastIterationNumber - 1;
+                var latestChildArray = latestChild.partIterations[lastIteration].components;
+                if (latestChildArray.length !== 0) {
+
+                    $(currentLine).addClass('product_title');
```

Comment:
use addClass only once (can add multiple class in a single call), same for removeClass